src/anderson.jl

```diff
@@ -77,13 +77,13 @@
     # xn+1 = xn+beta*f(xn) is convergent, but the convergence
     # criterion is not guaranteed
     return SolverResults("Anderson m=$m β=$β",
-                         x0, copy(xs[:,1]), maximum(abs,fx),
+                         x0, reshape(xs[:,1], size(x0)...), maximum(abs,fx),
```

Comment:
Is there a reason we had a `copy` of `xs` before. This call to `reshape` will return a view into `xs` for Array (perhaps other AbstractArray).

I'm not claiming this is a problem, just asking for clarification on the semantics so we know this is ok to do.

Reply:
No, it will return a view into a newly allocated vector `xs[:,1]`. The `copy` was needed there before to retrieve this newly allocated vector and not a  subarray - because of `@views` in line
https://github.com/JuliaNLSolvers/NLsolve.jl/blob/master/src/anderson.jl#L4
`xs[:,1]` alone would have returned a view into the matrix `xs`. Since I moved `@views` to lines
https://github.com/devmotion/NLsolve.jl/blob/426c3c2698697f0a7c5054720706855aae9461b9/src/anderson.jl#L30
and
https://github.com/devmotion/NLsolve.jl/blob/426c3c2698697f0a7c5054720706855aae9461b9/src/anderson.jl#L27
(these are the only other places where it caused views), I could get rid of `copy` in the results and just had to reshape it to the dimensions of the input.

Reply:
By the way, for the same reason `copy` is used in lines
https://github.com/JuliaNLSolvers/NLsolve.jl/blob/master/src/anderson.jl#L43-L44
and
https://github.com/JuliaNLSolvers/NLsolve.jl/blob/master/src/anderson.jl#L61
and
https://github.com/JuliaNLSolvers/NLsolve.jl/blob/master/src/anderson.jl#L72.